drv/minibar-seq-server/src/main.rs

```diff
@@ -126,6 +152,48 @@ impl idl::InOrderSequencerImpl for ServerImpl {
 
         Ok(state == DeviceState::RunningUserDesign)
     }
+
+    fn sled_power_status(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<Reg::VBUS_SLED::StateEncoded, RequestError<MinibarSeqError>>
+    {
+        self.get_sled_power_status()
+            .map_err(MinibarSeqError::from)
+            .map_err(RequestError::from)
+    }
+
+    fn sled_powered(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<bool, RequestError<MinibarSeqError>> {
+        let state = self
+            .get_sled_power_status()
+            .map_err(MinibarSeqError::from)
+            .map_err(RequestError::from)?;
+
+        Ok(state == Reg::VBUS_SLED::StateEncoded::Enabled)
+    }
+
+    fn sled_power_enable(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<(), RequestError<MinibarSeqError>> {
+        self.sled_power_control(true)
+            .map_err(MinibarSeqError::from)
+            .map_err(RequestError::from)?;
+        Ok(())
```

Comment:
Could you just return this result directly?
```suggestion
            .map_err(RequestError::from)
```
(also applicable below)